aopy/visualization/base.py

```diff
@@ -1163,23 +1163,60 @@ def plot_trajectories(trajectories, bounds=None, ax=None, **kwargs):
         ax (plt.Axis, optional): axis to plot the targets on
         kwargs (dict): keyword arguments to pass to the plt.plot function
    '''
+
     if ax is None:
         ax = plt.gca()
 
-    # Plot in 3D, fall back to 2D
+    # Plot in 3D, fall back to 2D and check axes
     try:
         ax.set_zlabel('z')
-        for path in trajectories:
-            ax.plot(*path.T, **kwargs)
+        for traj in trajectories:
+            ax.plot(*traj.T, **kwargs)
         ax.set_box_aspect((1, 1, 1))
+        ax.set_xlabel('x')
+        ax.set_ylabel('y')
     except:
-        for path in trajectories:
-            ax.plot(path[:, 0], path[:, 1], **kwargs)
+        stacked = np.vstack(trajectories)
+        zero_cols = np.all(stacked==0, axis=0)
+        
+        if stacked.shape[1]>2:
+
+            if not zero_cols.any():
+                warnings.warn("Axis is unclear for 3D data (no zero columns). Plots assume data of interest is in xy-plane.", stacklevel=2)
+                zero_col_idx = [2] # assume z-axis data is unwanted
+                col_to_remove = zero_col_idx[0] # tag z-axis data for removal
+            else:
+                zero_col_idx = np.where(zero_cols)[0]
+                
+                if len(zero_col_idx) > 1: # 1D data
+                    col_to_remove = zero_col_idx[1] # tag the 2nd all-zero axis for removal
+                else: # 2D data
+                    col_to_remove = zero_col_idx[0] # tag the only all-zero axis for removal
+                    
+            flattened = [traj[:, [i for i in range(3) if i != col_to_remove]] for traj in trajectories]
+            flattened = np.array(flattened, dtype=object)
+
+            if np.all(np.isin(2, zero_col_idx)):
+                ax.set_xlabel('x')
+                ax.set_ylabel('y')
+            elif np.all(np.isin(1, zero_col_idx)):
+                ax.set_xlabel('x')
+                ax.set_ylabel('z')
+            else:
+                ax.set_xlabel('z')
+                ax.set_ylabel('y')
```

Comment:
since this code is the same for both functions can you make it its own helper function that transforms 3d data into 2d data plus an x and y label?